shell/platform/darwin/ios/framework/Source/FlutterPlatformViews_Internal.mm

```diff
@@ -447,3 +451,63 @@ - (void)clipPath:(const SkPath&)path matrix:(const SkMatrix&)matrix {
 }
 
 @end
+
+@interface FlutterClippingMaskViewPool ()
+
+@property(assign, nonatomic) NSUInteger capacity;
+@property(retain, nonatomic) NSMutableArray<FlutterClippingMaskView*>* pool;
```

Comment:
Would this retain the `FlutterClippingMaskView` even after the `ChildClippingView` is dealloc'd?
Would `NSPointerArray` with `NSPointerFunctionsWeakMemory` (I think zeroes it out with MRC but worth testing) work better here?  If you could reliably know the mask views in the pool are used (because they are retained), then you could get rid of `recycleMaskViews` and during insert  `compact`, then insert at `count` if there was capacity.

Reply:
Well never mind, retaining it after the `ChildClippingView` is dealloc'd is the whole point of this PR...
Can you confirm that deallocating `FlutterPlatformViewsController` releases the pool and the mask views?